aws/resource_aws_ec2_client_vpn_authorization_rule.go

```diff
@@ -0,0 +1,277 @@
+
+	input := &ec2.RevokeClientVpnIngressInput{
+		ClientVpnEndpointId: aws.String(d.Get("client_vpn_endpoint_id").(string)),
+		TargetNetworkCidr:   aws.String(d.Get("target_network_cidr").(string)),
+		RevokeAllGroups:     aws.Bool(d.Get("authorize_all_groups").(bool)),
+	}
+	if v, ok := d.GetOk("access_group_id"); ok {
+		if s, ok := v.(string); ok && s != "" {
+			input.AccessGroupId = aws.String(s)
+		}
+	}
+
+	log.Printf("[DEBUG] Revoking Client VPN authorization rule %q", d.Id())
+	err := deleteClientVpnAuthorizationRule(conn, input)
+	if err != nil {
+		return fmt.Errorf("error revoking Client VPN authorization rule %q: %w", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceAwsEc2ClientVpnAuthorizationRuleImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	endpointID, targetNetworkCidr, accessGroupID, err := tfec2.ClientVpnAuthorizationRuleParseID(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	d.Set("client_vpn_endpoint_id", endpointID)
+	d.Set("target_network_cidr", targetNetworkCidr)
+	d.Set("access_group_id", accessGroupID)
+	return []*schema.ResourceData{d}, nil
+}
+
+func deleteClientVpnAuthorizationRule(conn *ec2.EC2, input *ec2.RevokeClientVpnIngressInput) error {
+	id := tfec2.ClientVpnAuthorizationRuleCreateID(aws.StringValue(input.ClientVpnEndpointId), aws.StringValue(input.TargetNetworkCidr), aws.StringValue(input.AccessGroupId))
+
+	_, err := conn.RevokeClientVpnIngress(input)
+	if isAWSErr(err, tfec2.ErrCodeClientVpnAuthorizationRuleNotFound, "") {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	_, err = ClientVpnAuthorizationRuleRevoked(conn, id)
+
+	return err
+}
+
+func ClientVpnAuthorizationRuleAuthorized(conn *ec2.EC2, authorizationRuleID string) (*ec2.AuthorizationRule, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{ec2.ClientVpnAuthorizationRuleStatusCodeAuthorizing},
+		Target:  []string{ec2.ClientVpnAuthorizationRuleStatusCodeActive},
+		Refresh: ClientVpnAuthorizationRuleStatus(conn, authorizationRuleID),
+		Timeout: waiter.ClientVpnAuthorizationRuleActiveTimeout,
+	}
+
+	outputRaw, err := stateConf.WaitForState()
+
+	if output, ok := outputRaw.(*ec2.AuthorizationRule); ok {
+		return output, err
+	}
+
+	return nil, err
+}
+
+func ClientVpnAuthorizationRuleRevoked(conn *ec2.EC2, authorizationRuleID string) (*ec2.AuthorizationRule, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{ec2.ClientVpnAuthorizationRuleStatusCodeRevoking},
+		Target:  []string{},
+		Refresh: ClientVpnAuthorizationRuleStatus(conn, authorizationRuleID),
+		Timeout: waiter.ClientVpnAuthorizationRuleRevokedTimeout,
+	}
+
+	outputRaw, err := stateConf.WaitForState()
+
+	if output, ok := outputRaw.(*ec2.AuthorizationRule); ok {
+		return output, err
+	}
+
+	return nil, err
+}
+
+// ClientVpnAuthorizationRuleStatus fetches the Client VPN authorization rule and its Status
+// TODO: This should be in the waiters package, but has a dependency on isAWSErr()
```

Comment:
Looks like this TODO may have been satisfied with the temporary function addition in `aws/internal/service/ec2`.